duckduckgo/Android
# Autofill security update

Task/Issue URL: https://app.asana.com/0/414730916066338/1200713984364738
Tech Design URL: 
CC: 

**Description**:
This release updates autofill to 3.2.1, removes a few interfaces that we don't use anymore and checks that to use some interfaces the user has to be in duckduckgo.com/email.  

**Steps to test this PR**:
Make sure you are up to date with autofill so execute `git submodule update submodules/autofill`

Before testing, go to `EmailJavascriptInterface` and add the following:
1. Before the line 44 add `Timber.d("TEST isSignedIn")`
1. Before the line 53 add 1. Before the line 44 add `Timber.d("TEST storeCredentials")`

**Autofill is correctly updated**
1. Checkout branch
1. Run git submodule
1. Confirm autofill commit hash is `8a08e934d538e06735df0b13e31f40ace4517b46`

**Test cases**
1. Install the app
1. Go to `github.com`
1. In the logcat you should NOT see `TEST isSignedIn`
1. Go to `https://duckduckgo.com/email/login` either via the browser or via `Settings/Email Protection/I already have a Duck Address`
1. In the logcat you should see `TEST isSignedIn`
1. Sign in with your account
1. In the logcat you should see `TEST storeCredentials`.
1. Go to `github.com` and make sure you can see Dax and clicking on it shows the tooltip.
1. Click on the tooltip buttons and make sure they work as supposed to.

**Cannot execute code from host**
1. Go to `github.com`
1. In desktop Chrome or Brave go to `chrome://inspect/`
1. Inspect your current tab which should be `github.com`
1. In the console execute `window.EmailInterface.storeCredentials("test", "test", "test")`
1. The console should return undefined and in the logcat you should not see `TEST storeCredentials`
1. In the console execute `window.EmailInterface.isSignedIn()`
1. The console should return "" and in the logcat you should not see `TEST isSignedIn`
1. Go to `duckduckgo.com/email`
1. Back into the inspect tab in the console 
1. In the console execute `window.EmailInterface.isSignedIn()`
1. The console should return "" and in the logcat you should see `TEST isSignedIn`
1. In the console execute `window.EmailInterface.storeCredentials("test", "test", "test")`
1. The console should return undefined and in the logcat you should see `TEST storeCredentials`
1. At this point you credentials will be wrong but nothing else to test!


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
